File: v2/src/main/java/com/api/v2/doctors/services/DoctorRegistrationService.java
package com.api.v2.doctors.services;

import com.api.v2.doctors.requests.DoctorRegistrationDto;
import com.api.v2.doctors.responses.DoctorResponseDto;
import org.springframework.http.ResponseEntity;
import reactor.core.publisher.Mono;

public interface DoctorRegistrationService {
    Mono<ResponseEntity<DoctorResponseDto>> register(DoctorRegistrationDto registrationDto);
}

File: v2/src/main/kotlin/com/api/v2/doctors/services/DoctorRegistrationService.kt
package com.api.v2.doctors.services

import com.api.v2.doctors.DoctorResponseDTO
import com.api.v2.people.PersonRegistrationDTO
import org.springframework.http.ResponseEntity

interface DoctorRegistrationService {
    fun register(licenseNumber: String, personDTO: PersonRegistrationDTO): ResponseEntity<DoctorResponseDTO>
}